lib/travis/api/app/endpoint/jobs.rb

```diff
@@ -80,8 +80,12 @@ class Jobs < Endpoint
         elsif (!resource || resource.archived?)
           # the way we use responders makes it hard to validate proper format
           # automatically here, so we need to check it explicitly
-          if accepts?('text/plain')
-            archived_log_path = archive_url("/jobs/#{params[:job_id]}/log.txt")
+          if accepts?('text/plain') || request.user_agent.to_s.start_with?('Travis')
```

Comment:
I think there may be something wrong with `accepts?` although I haven't been able to find out how/why, which is why I've added this _really awful_ special casing of `^Travis` value of `User-Agent` instead of having to patch & release https://github.com/travis-ci/travis.rb.